gensound/transforms.py

```diff
@@ -190,31 +190,50 @@ def realise(self, audio):
 class Fade(Transform):
     min_fade = -50
     
-    def __init__(self, is_in=True, duration=3e3):
-        self.is_in = is_in
+    def __init__(self, duration=None, curve=None, is_in=True, degree=None):
+        assert curve in (None, "linear", "polynomial")
+        # default setting for backwards compatibility (allowing usage of both Fade() and FadeIn/FadeOut()
+        if curve == None:
+            curve = "linear"
+        if duration == None:
+            duration = 3e3
+        if degree == None:
+            degree = 1
```

Comment:
Python already has a mechanism to provide default values for arguments, use it! :)

Reply:
I thought about it and preferred this way! I'll explain, and if you aren't convinced - I'll change. 
Although `Fade` is not the preferred way, I still want the have it ready to be called by the user.
So, the defaults values have to be in three different places: `Fade()`, `FadeIn`,`FadeOut`, and it doesn't look as a good practice to me - any change later have to be manually in 3 spots.  
So I have set all the defaults to `None` in all those places, (rather then no default at all, that way the user is not forced to specify any parameter if he doesn't wish to), and, since all of them initialize in the same place, any parameter which is None, meaning the user hasn't set it - will be set to a default value.